woocommerce/wc-admin
# Fixing button state logic for payment gateways

Fixes woocommerce/woocommerce-admin#7184

Reworking logic around various buttons and states with the payment gateways in remote payments. As per the original issue, these are the states and expected behavior:

### Core gateways (CODS & BACS)
**Default**
If setup required, display `Setup` button. Otherwise display `Enable` button.

**Enabled**
Display `Manage` button, and list in the _Enabled gateways_ list.

**Note:** These methods do not have a "finish setup" state, even if setup is required.


### External gateways with setup
**Default**
Display with `Setup` button in _additional gateways_ list.

**Installed & not configured**
Display with _Finish Setup_ UI and `Finish Setup` button.

**Installed & configured**
Display with `Manage` button, in _enabled_ list if enabled, or _additional_ otherwise.


### Screenshots

![image](https://user-images.githubusercontent.com/444632/122289482-9e8f8c00-cea7-11eb-9e4a-037ee7c787c3.png)


### Detailed test instructions:

-   Ensure remote payments is enabled, and this branch is in use
- Navigate to payment task
-   Run through the enabling/setup process with COD and BACS, and ensure the button states are as expected from description above.
- Test an external gateway that requires setup, such as Razorpay. You have to use the modified version (see [PR list](woocommerce/woocommerce#32258)) for the correct behavior.
- Test WCPay card states, which should mirror the above, but with a _Get started_ button as the default state. You'll also need to use our [modified branch](https://github.com/Automattic/woocommerce-payments/pull/2014).
